queries.sql

```diff
@@ -9,4 +9,80 @@ SELECT name FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name NOT LIKE '%Gabumon%';
 SELECT name FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
 
+/*
+Inside a transaction update the animals table by setting the species column to unspecified. 
+Verify that change was made. Then roll back the change and verify that the species columns 
+went back to the state before the transaction.
+*/
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+/*
+Inside a transaction:
+    Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+    Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+    Commit the transaction.
+    Verify that change was made and persists after commit.
+*/
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+
+/*
+Inside a transaction:
+    Delete all records in the animals table, then roll back the transaction.
+    After the rollback verify if all records in the table are still there. 
+*/
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+/* 
+    Inside a new transaction, delete all animals born after Jan 1st, 2022.
+    Create a save point for the transaction.
+*/
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT savepoint_name;
+
+/* 
+    Update all animals' weight to be their weight multiplied by -1. 
+    Verify that change was made. Then roll back to the save point.
+    After the rollback, verify that all records have positive weight values again.
+    Commit transaction.
+*/
+
+UPDATE animals SET weight_kg = weight_kg * -1;
+SELECT * FROM animals;
+ROLLBACK TO savepoint_name;
+SELECT * FROM animals;
```

Comment:
After you rollback to the savepoint
*  Kindly `UPDATE` all animals' weights that are negative to be their weight multiplied by -1 as per project requirements.

Then commit the transaction

Reply:
well resolved! ✔️